edit_course/course_forms.py

```diff
@@ -88,6 +89,9 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
             instance=self.instance.active_model_solution_reveal_rule,
             prefix='model_solution',
         )
+        self.fields['model_answer'].queryset = CourseChapter.objects.filter(
+            course_module__course_instance=self.instance.course_instance,
+        )
```

Comment:
In addition, I was thinking that the model could validate that the model solution chapter belongs to the same course.

https://github.com/apluslms/a-plus/blob/3893d834479b0d8f4a301942ae98a894006b39eb/course/models.py#L1361-L1377

I can try that myself.

Reply:
Done here: https://github.com/apluslms/a-plus/commit/a180fbf1cff90743776fadbfd6d51ba15823f046